README.md

```diff
@@ -1,9 +1,170 @@
+| `exec-forward.pod.kubernetes.io/pre-connect` | A JSON formatted list of commands executed before establishing a port-forwarding connection |
+| `exec-forward.pod.kubernetes.io/post-connect` | A JSON formatted list of commands executed after establishing a port-forwarding connection |
+| `exec-forward.pod.kubernetes.io/command` | A single JSON formatted command ran after `post-connect` |
+
+#### Command
+
+##### Object
+
+The command object represents an exec command and associated configuration to be executed by the plugin during the port-forwarding lifecycle.
+
+| Attribute | Description | Optional |
+| --- | --- | --- |
+| `id` | A unique identifier that can be used in subsequent commands to reference a previous command's output | `true` |
+| `command` | The command to run as an array of strings | `false` |
+| `interactive` | Whether the command should be run in interactive mode. Default is `false`. Note: the main `command` is always run in interactive mode | `true` |
+| `name` | The display name for the command, shown during execution | `false` |
+
+##### Rendering
+
+Exec commands are [Go templated](https://pkg.go.dev/text/template) using inputs from the CLI, the `args` annotation, configuration from the port-forwarding connection and previous command outputs.
+
+| Namespace | Description | Example |
+|---|---|---|
+| `.Args` | Arguments read from the `args` annotation and overridden using the `--arg\|-a` CLI flags | `{{.Args.username}}` |
+| `.Outputs` | Stdout and Stderr from previously ran commands, stored by command `id` | `{{.Outputs.foo.Stdout}}` |
+| `.Config` | Port-forwarding configuration | `{{.Config.LocalPort }}` |
+
+##### Template functions
+
+In addition to the following, all standard Go templating functions are available for use
+
+| Name | Description |
+| --- | --- |
+| `sensitive` | Replaces the passed value with `********` when printing to console |
+| `trim` | Removes white space from the beginning and end of a string, useful when piping to other template functions |
+
+## Examples
+
+Connect to an AWS RDS database using a generated password
+
+```yaml
+kind: Pod
+metadata:
+  name: db
+  annotations:
+    exec-forward.pod.kubernetes.io/args: '{"username":"read"}'
+    exec-forward.pod.kubernetes.io/pre-connect: |
+      [{
+        "command": [
+          "aws",
+          "rds",
+          "generate-db-auth-token",
+          "--host",
+          "...rds.amazonaws.com",
+          "--port",
+          "5432",
+          "--username",
+          "{{.Args.username}}"
+        ],
+        "id": "password",
+        "name": "Generate temporary password"
+      }]'
+    exec-forward.pod.kubernetes.io/command: |
+      {
+        "command":[
+          "psql",
+          "postgres://{{.Args.username}}:{{ trim .Outputs.password.Stdout | urlquery | sensitive }}@localhost:{{.Config.LocalPort}}/db"
+        ]
+      }
+spec:
+  containers:
+    image: alpine/socat
+    - command:
+      - socat
```

Comment:
Worth explaining the use of `socat` a bit more here and how it's serving as an active forwarder in Kubernetes.